### PR TITLE
[17.10] add 17.10 changelog

### PR DESCRIPTION
**Pull request by vieux**

### CODE REVIEW

**Comment by vieux** [comment]
ping @thaJeztah @fcrisciani @tonistiigi

File: CHANGELOG [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+* Move output of `docker stack rm` to stdout [docker/cli#491](https://github.com/docker/cli/pull/491)
+* Remove deprecated `--enable-api-cors` flag [moby/moby#34821](https://github.com/moby/moby/pull/34821)
+* Use nature sort secrets and configs in cli [docker/cli#307](https://github.com/docker/cli/pull/307)
+* Use non-detached mode as default for `docker service` commands [docker/cli#525](https://github.com/docker/cli/pull/525)
+* Set APIVersion on the client, even when Ping fails [docker/cli#546](https://github.com/docker/cli/pull/546)
+
+### Logging
+
+* Move pkg/jsonlog to be a subpackage of the single consumer [moby/moby#34946](https://github.com/moby/moby/pull/34946)
```
Comment (thaJeztah) [inline CHANGELOG]: This feels like an internal change, not sure it's worth mentioning

File: CHANGELOG [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+- Fixed `raw` mode splunk logger [moby/moby#34520](https://github.com/moby/moby/pull/34520)
```
Comment (thaJeztah) [inline CHANGELOG]: This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/233, so not new in 17.10

File: CHANGELOG [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+### Networking
+
+* Ensure Host Network Service exists [moby/moby#34928](https://github.com/moby/moby/pull/34928)
```
Comment (thaJeztah) [inline CHANGELOG]: This is a bug fix (Windows only); https://github.com/moby/moby/issues/34920

File: CHANGELOG [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* Enabling ILB/ELB on windows using per-node, per-network LB endpoint. [moby/moby#34674](https://github.com/moby/moby/pull/34674)
+
+### Runtime
+
+* LCOW: Add UVM debugability by grabbing logs before tear-down [moby/moby#34846](https://github.com/moby/moby/pull/34846)
+* LCOW: Fix comment in graphdriver code [moby/moby#34891](https://github.com/moby/moby/pull/34891)
```
Comment (thaJeztah) [inline CHANGELOG]: This is a comment-only change in the source; we can drop this one

File: CHANGELOG [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+* LCOW: Prepare work for bind mounts [moby/moby#34258](https://github.com/moby/moby/pull/34258)
+* LCOW: Support for docker cp, ADD/COPY on build [moby/moby#34252](https://github.com/moby/moby/pull/34252)
+* LCOW: VHDX boot to readonly. [moby/moby#34754](https://github.com/moby/moby/pull/34754)
```
Comment (thaJeztah) [inline CHANGELOG]: Perhaps we should combine all the LCOW lines to something like: "Add initial support for Linux Containers On Windows (LCOW)" - or better wording

@friism ^^

File: CHANGELOG [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+- Fix volume inspect with empty ID [moby/moby#34770](https://github.com/moby/moby/pull/34770)
+- Fix variable shadowing causing LayersSize to be reported as 0 [moby/moby#34826](https://github.com/moby/moby/pull/34826)
```
Comment (thaJeztah) [inline CHANGELOG]: This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/232, so not a new change

File: CHANGELOG [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+* Enabling ILB/ELB on windows using per-node, per-network LB endpoint. [moby/moby#34674](https://github.com/moby/moby/pull/34674)
+
+### Runtime
+
+* LCOW: Add UVM debugability by grabbing logs before tear-down [moby/moby#34846](https://github.com/moby/moby/pull/34846)
+* LCOW: Fix comment in graphdriver code [moby/moby#34891](https://github.com/moby/moby/pull/34891)
+* LCOW: Prepare work for bind mounts [moby/moby#34258](https://github.com/moby/moby/pull/34258)
+* LCOW: Support for docker cp, ADD/COPY on build [moby/moby#34252](https://github.com/moby/moby/pull/34252)
+* LCOW: VHDX boot to readonly. [moby/moby#34754](https://github.com/moby/moby/pull/34754)
+- Fix volume inspect with empty ID [moby/moby#34770](https://github.com/moby/moby/pull/34770)
```
Comment (thaJeztah) [inline CHANGELOG]: Looks like this was cherry-picked into 17.09 https://github.com/docker/docker-ce/commit/7034626ac60e8da8a15e874662e0fb9635a7fefc#diff-65244c7ec61daa507c915173882e5de3R215 (only in components/cli, but it looks like a client fix)

/cc @dnephin

File: CHANGELOG [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+- Fix variable shadowing causing LayersSize to be reported as 0 [moby/moby#34826](https://github.com/moby/moby/pull/34826)
+* Volume: evaluate symlinks before relabeling mount source [moby/moby#34792](https://github.com/moby/moby/pull/34792)
```
Comment (thaJeztah) [inline CHANGELOG]: Should we mention `SELinux` here?

File: CHANGELOG [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+* Set selinux label on local volumes from mounts API [moby/moby#34684](https://github.com/moby/moby/pull/34684)
```
Comment (thaJeztah) [inline CHANGELOG]: This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/241, so not new

File: CHANGELOG [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+* Close pipe in overlay2 graphdriver [moby/moby#34863](https://github.com/moby/moby/pull/34863)
```
Comment (thaJeztah) [inline CHANGELOG]: This was back ported to 17.09 through https://github.com/docker/docker-ce/pull/245, so not new

File: CHANGELOG [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+- Fixing ‘docker cp’ to allow new target file name in a host symlinked directory [moby/moby#31993](https://github.com/moby/moby/pull/31993)
+
+### Swarm Mode
+
+* Use limit with `docker swarm init --force-new-cluster` [moby/moby#34881](https://github.com/moby/moby/pull/34881)
```
Comment (thaJeztah) [inline CHANGELOG]: Perhaps

```Markdown
* Produce an error if `docker swarm init --force-new-cluster` is executed on worker nodes [moby/moby#34881](https://github.com/moby/moby/pull/34881)
```

File: CHANGELOG (hunk 7 above)
Comment (dnephin) [reply CHANGELOG]: Yes, I agree this fix is already in 17.09.

**Comment by thaJeztah** [comment]
https://github.com/docker/cli/pull/565 was just merged, and needs an entry if we didn't cut off the 17.10 branch yet;

```Markdown
* Change the default output format for `docker container stats` to show `CONTAINER ID` and `NAME` [docker/cli#565](https://github.com/docker/cli/pull/565)
+ Add `--no-trunc` flag to `docker container stats` [docker/cli#565](https://github.com/docker/cli/pull/565)
```

**Comment by dnephin** [comment]
`docker trust` is in experimental for 17.10 , right? Should we mention it?

**Comment by thaJeztah** [comment]
Ah, yes, that should definitely be added, thanks!

**Comment by vieux** [comment]
@thaJeztah I addressed most of your comments, feel free to edit the rest directly if you want, I wasn't sure what to do, otherwise, I think it's good for RC1

**Comment by vieux** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

**Comment by andrewhsu** [comment]
cc @JimGalasyn

**Comment by mstanleyjones** [comment]
@andrewhsu FYI CE stuff is me and @gbarr01 :)

**Pull request merged**